opensearch_single_kernel/managers/lock.py

```diff
@@ -0,0 +1,507 @@
+                        payload={"unit-name": self.state.opensearch_unit.unit_name},
+                    )
+                except OpenSearchHttpError as e:
+                    if e.response_code == 409 and "document already exists" in e.response_body.get(
+                        "error", {}
+                    ).get("reason", ""):
+                        # Document already created
+                        self.logger.debug(
+                            "[Node lock] Another unit acquired OpenSearch lock while this unit attempted "
+                            "to acquire lock"
+                        )
+                        return False
+                    else:
+                        self.logger.exception("Error creating OpenSearch lock document")
+                        # in this case, try to acquire peer databag lock as fallback
+                        return super().acquired
+                else:
+                    # Ensure write was successful on all nodes
+                    # "It is important to note that this setting [`wait_for_active_shards`] greatly
+                    # reduces the chances of the write operation not writing to the requisite
+                    # number of shard copies, but it does not completely eliminate the possibility,
+                    # because this check occurs before the write operation commences. Once the
+                    # write operation is underway, it is still possible for replication to fail on
+                    # any number of shard copies but still succeed on the primary. The `_shards`
+                    # section of the write operation’s response reveals the number of shard copies
+                    # on which replication succeeded/failed."
+                    # from
+                    # https://www.elastic.co/guide/en/elasticsearch/reference/8.13/docs-index_.html#index-wait-for-active-shards
+                    if response["_shards"]["failed"] > 0:
+                        self.logger.error("Failed to write OpenSearch lock document to all nodes.")
+                        self.logger.debug(
+                            "[Node lock] Deleting OpenSearch lock after failing to write to all nodes"
+                        )
+                        # Delete document id 0
+                        self.opensearch_client.request(
+                            "DELETE",
+                            endpoint=f"/{self.OPENSEARCH_INDEX}/_doc/0?refresh=true",
+                            host=host,
+                            alt_hosts=self.state.alt_hosts,
+                            retries=10,
+                        )
+                        self.logger.debug(
+                            "[Node lock] Deleted OpenSearch lock after failing to write to all nodes"
+                        )
+                        return False
+
+                    # This unit has OpenSearch lock
+                    unit = self.state.opensearch_unit.unit_name
```

Comment:
```suggestion
                    unit = self.state.unit_name
 
```
self.state.opensearch_unit.unit_name does not exist.